src/Kiota.Builder/Extensions/StringExtensions.cs

```diff
@@ -11,14 +12,21 @@ public static string ToFirstCharacterLowerCase(this string input)
             => string.IsNullOrEmpty(input) ? input : $"{char.ToLowerInvariant(input.FirstOrDefault())}{input[1..]}";
         public static string ToFirstCharacterUpperCase(this string input)
             => string.IsNullOrEmpty(input) ? input : Char.ToUpperInvariant(input.FirstOrDefault()) + input[1..];
-        public static string ToCamelCase(this string name)
+
+        /// <summary>
+        /// Converts a string delimited by a symbol to camel case
+        /// </summary>
+        /// <param name="input">The input string</param>
+        /// <param name="delimiters">The delimiters to use when converting to camel case. If none is given, defaults to '-'</param>
+        /// <returns>A camel case string</returns>
+        public static string ToCamelCase(this string input, params string[] delimiters)
         {
-            if(string.IsNullOrEmpty(name)) return name;
-            var chunks = name.Split('-', StringSplitOptions.RemoveEmptyEntries);
-            var identifier = chunks[0] + string.Join(string.Empty, chunks.Skip(1)
-                                                                .Select(s => ToFirstCharacterUpperCase(s)));
-            return identifier;
+            if (string.IsNullOrEmpty(input)) return input;
+            delimiters = delimiters.Any() ? delimiters : new[] { "-" };
+            var chunks = input.Split(delimiters, StringSplitOptions.RemoveEmptyEntries);
+            return string.Join(string.Empty, chunks.Select((c, i) => i == 0 ? c : c.ToFirstCharacterUpperCase()));
```

Comment:
```suggestion
            return chunks[0] + string.Join(string.Empty, chunks.Skip(1)
                                                                .Select(s => ToFirstCharacterUpperCase(s)))
```

(easier to read and avoids testing the index on each select callback)

Reply:
If I did that, I'd have to check the chunks length.